glean-core/src/upload/mod.rs

```diff
@@ -360,6 +365,18 @@ impl PingUploadManager {
             return;
         }
 
+        {
+            let in_flight = self.in_flight.read().unwrap();
+            if in_flight.contains_key(document_id) {
+                log::warn!(
+                    "Attempted to enqueue an in-flight ping {} at {}.",
+                    document_id,
+                    path
+                );
+                return;
```

Comment:
Suggestion: This `return;` could be a big deal. We should have a health metric for it and a place on the monitoring dashboards for it.

Reply:
I second this. But I think if we file a bug and drop a TODO here we don't need to block on this